check_tools/config-update.py

```diff
@@ -0,0 +1,65 @@
+import sys, os
+import re
+import subprocess
+
+def split_into_pairs(l):
+    for i in range(0, len(l), 2):
+        yield l[i:i + 2]
+
+def main(file):
+    file_h = open(file, 'r+')
+    file   = file_h.read()
+    snippet_indices = [m.start() for m in re.finditer('```', file)]
+
+    blocks = {}
+    for i in range(0, int(len(snippet_indices) / 2)):
+        # Need to rerun on every loop as the indices change each iteration
+        snippet_indices = [m.start() for m in re.finditer('```', file)]
+        ranges = list(split_into_pairs(snippet_indices))
+        start  = ranges[i][0]
+        end    = ranges[i][1]
+
+        print(snippet_indices)
+
+        try:
+            blocks[i] = file[start : end + 3]
+            if ('Name: ' in blocks[i]):
+                lib = blocks[i].split('Name: ')[1].split('.')[0]
+                print("=================   %s   =================" % lib)
+                out = subprocess.check_output(["mbed", "compile", "--config", "-v", "--prefix", lib]).decode()
```

Comment:
You probably want:

```suggestion
                out = subprocess.check_output(["mbed", "compile", "--config", "-v", "--prefix", lib]).encode('utf-8')
```

For Py2/Py3 compat.

Reply:
The subprocess output in Python3 is a Bytes object that I cannot encode. I had added  the decode initially to allow it to run in both Python2/3 (although I did run the following error in nvstore in Python2):

```
'ascii' codec can't decode byte 0xe2 in position 996: ordinal not in range(128)
```

Reply:
https://github.com/ARMmbed/mbed-os/pulls?q=is%3Apr+is%3Aclosed+author%3Acmonr

A couple of my recent PRs were Python 2/3 compat related if you want to take a look.

Imo, it might be better if this works with Py3 if you had to choose one. 

Also, I _might_ be confusing `encode` and `decode`.

Reply:
It would be good to know _what_ causes the occasional convert error.